debian/control

```diff
@@ -603,37 +603,6 @@ Description: Enhanced system logging daemon (Kafka destination, based on librdka
  This package provides a native Kafka destination, written entirely in the C programming
  language, based on the librdkafka client library.
 
-Package: syslog-ng-mod-extra
```

Comment:
When I created this package I was afraid that adding these plugins will add unwanted dependencies as all the scl files are loaded automatically and every one of them complains if they could not find something. Also at that time, the json plugin was optional (my opinion is that it still should be as the 99% of the logs are still simple text entries) and quite a few of them were depending on the json module too.

In my opinion, that is not enough that the user got a nice error message if that error message complains about an actually unwanted dependency of an unwanted feature.

Reply:
These don't add "dependencies" in the sense that syslog-ng would not depend on additional packages and would not even display error/warning messages at startup unless the user is explicitly using those pieces (if it does, it's a bug!). These were improved recently in the syslog-ng codebase (https://github.com/syslog-ng/syslog-ng/pull/3015, merged for 3.26).

So, as a summary, syslog-ng does not complain about an unwanted dependency of an unwanted feature. e.g. if syslog-ng-mod-http() is not installed, the slack() destination would complain about the missing http() module with this error message, but only if slack() is actually used in the configuration!

```
@requires http "The slack() driver depends on the syslog-ng http module, please install the syslog-ng-mod-http (Debian & derivatives) or the syslog-ng-http (RHEL & co) package"
```

json support is different (as it is pretty widely used to encode log messages), my goal has been to make that mandatory for all installations, because without it, we are limited to the ancient, text-only syslog use-cases, keeping the perception that syslog-ng is only good for those use-cases (which it isn't). Just think about elasticsearch, graylog, loggly, slack... 

The json dependencies were pulled into the "core" package more than a year ago (2017 in our git logs, end of 2018 in the official debian packaging), and RHEL/Fedora followed suit.

The JSON format is increasingly needed for core use-cases as well, for instance we are trying to add a syslog-ng() destination, that encodes all name-value pairs and sends them on. The format that is used here is JSON. Without JSON in the "core" package, we would be unable to promote this format. 

With that said, I can imagine a syslog-ng deployment _without_ json, but in my mind that should not have SCL at all. So just avoid including "scl.conf" & do not ship the scl include files, then syslog-ng would work without JSON (at least as long until start depending on it in the core, which we avoided so far, but came up already). But then it should be easy to add "scl" on top of a traditional config file, to open up more recent use-cases.

I hope this explanation helps, let me know if you have further questions.